lib/engine/step/dividend.rb

```diff
@@ -10,7 +10,7 @@ class Dividend < Base
       ACTIONS = %w[dividend].freeze
 
       def actions(entity)
-        return [] if entity.company? || routes.empty?
+        return [] if entity.company? || @game.routes_revenue(routes).zero?
```

Comment:
Trigger the dividend step for games that have revenue that comes from other places beyond their trains.